src/cmd/tag.go

```diff
@@ -0,0 +1,235 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/opslevel/cli/common"
+	"github.com/opslevel/opslevel-go/v2023"
+	"golang.org/x/exp/slices"
+
+	"github.com/spf13/cobra"
+)
+
+type TaggableResourceFetchFunction func(id opslevel.ID) (any, error)
+
+var TaggableResourceFetchFunctions = map[opslevel.TaggableResource]TaggableResourceFetchFunction{
+	opslevel.TaggableResourceService:                func(id opslevel.ID) (any, error) { return getClientGQL().GetService(id) },
+	opslevel.TaggableResourceRepository:             func(id opslevel.ID) (any, error) { return getClientGQL().GetRepository(id) },
+	opslevel.TaggableResourceTeam:                   func(id opslevel.ID) (any, error) { return getClientGQL().GetTeam(id) },
+	opslevel.TaggableResourceUser:                   func(id opslevel.ID) (any, error) { return getClientGQL().GetUser(string(id)) },
+	opslevel.TaggableResourceDomain:                 func(id opslevel.ID) (any, error) { return getClientGQL().GetDomain(string(id)) },
+	opslevel.TaggableResourceSystem:                 func(id opslevel.ID) (any, error) { return getClientGQL().GetSystem(string(id)) },
+	opslevel.TaggableResourceInfrastructureresource: func(id opslevel.ID) (any, error) { return getClientGQL().GetInfrastructure(string(id)) },
+}
+
+type TaggableResourceFetchAliasFunction func(alias string) (any, error)
+
+var TaggableResourceFetchAliasFunctions = map[opslevel.TaggableResource]TaggableResourceFetchAliasFunction{
```

Comment:
Lets put a TODO here

// TODO: once we unifiy `GetService` and `GetServiceWithAlias` as well as the others we should beable to get rid of this code

Then lets cut a backlog ticket for fixing this

Reply:
@rocktavious filing that ticket and adding this // TODO comment now.

Reply:
@rocktavious https://github.com/OpsLevel/team-platform/issues/94